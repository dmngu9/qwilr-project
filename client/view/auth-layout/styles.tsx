import styled from '../../utils/styled-component';
import { media } from '../../utils/breakpoint';
import { flexCenter, borderBox } from '../../utils/mixins';

export const Background = styled.div`
    ${flexCenter()};
    height: 100vh;
    width: 100%;
    background: linear-gradient(20deg, ${props => props.theme.primaryColor}, #daa357);
`;

export const FormSection = styled.div`
    ${media.phone`
        width: 100%;
        padding: 32px 32px;
    `};

    display: initial;
    width: 400px;
    background-color: white;
    padding: 32px 48px;
    ${borderBox()};
`;
