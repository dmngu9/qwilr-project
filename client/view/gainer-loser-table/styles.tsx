import styled from '../../utils/styled-component';

export const Container = styled.div`
    margin-top: 48px;
`;

export const TableContainer = styled.div`
    width: 100%;
    margin-top: 16px;
`;

export const Header = styled.div`
    text-align: center;
    font-size: 18px;
    font-weight: 300;
    color: #3a3a3a;
    padding: 0 0 10px;
    text-decoration: underline;
`;
