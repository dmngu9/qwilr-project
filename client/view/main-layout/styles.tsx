import styled from '../../utils/styled-component';
import { media } from '../../utils/breakpoint';

export const Page = styled.div`
    ${media.phone`
        padding: 24px;
    `} padding: 48px;
`;
