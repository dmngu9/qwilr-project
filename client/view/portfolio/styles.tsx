import styled, { css, ThemedStyledProps, Theme } from '../../utils/styled-component';
import { Link, LinkProps } from 'react-router-dom';

import { media } from '../../utils/breakpoint';

type Props = ThemedStyledProps<LinkProps, Theme>;

export const Container = styled.div`
    margin-top: 48px;
`;

export const TableContainer = styled.div`
    width: 100%;
    margin-top: 16px;
`;

export const Header = styled.div`
    text-align: center;
    font-size: 18px;
    font-weight: 300;
    color: #3a3a3a;
    padding: 0 0 10px;
    text-decoration: underline;
`;

export const Total = styled.p`
    font-size: 14px;
    margin-top: 16px;
    margin-bottom: 16px;
    text-align: center;
`;

const linkMixins = () => css`
    color: ${(props: Props) => props.theme.primaryColor};

    &:active,
    &:hover {
        color: #e91b89;
    }
`;

export const BuyLink = styled(Link)`
    ${linkMixins()};
    margin-right: 16px;
`;

export const SellLink = styled(Link)`
    ${linkMixins()};
`;

export const RowCell = styled.p`
    ${media.phone`
        font-size: 12px;
    `};

    margin: 0;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    font-size: 18px;
`;

export const Buttons = styled.div`
    display: flex;
    justify-content: flex-start;
    align-items: center;
    flex-wrap: wrap;
`;
