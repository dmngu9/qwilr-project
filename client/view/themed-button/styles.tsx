import styled, { ThemedStyledProps, Theme } from '../../utils/styled-component';
import { borderBox } from '../../utils/mixins';

interface ExtraProps {
    complementary?: boolean;
}

type Props = ThemedStyledProps<ExtraProps, Theme>;

export const Button = styled.button`
    height: 32px;
    background-color: ${(props: Props) => (!props.complementary ? props.theme.primaryColor : 'white')};
    color: ${(props: Props) => (!props.complementary ? 'white' : props.theme.primaryColor)};
    ${borderBox()};
    padding: 0 8px;
    font-size: 14px;
    border: ${(props: Props) => (!props.complementary ? 'none' : `2px solid ${props.theme.primaryColor}`)};
    transition: background-color 0.3s ease-in-out, color 0.3s ease-in-out, border-color 0.3s ease-in-out;

    &:focus {
        outline: 0;
    }

    &:hover {
        cursor: pointer;
        background-color: ${(props: Props) => (!props.complementary ? '#e95a89' : 'white')};
        color: ${(props: Props) => (!props.complementary ? 'white' : '#e95a89')};
        border: ${(props: Props) => (!props.complementary ? 'none' : '2px solid #e95a89')};
    }

    &:active {
        background-color: ${(props: Props) => (!props.complementary ? '#e91B89' : 'white')};
        color: ${(props: Props) => (!props.complementary ? 'white' : '#e91B89')};
        border: ${(props: Props) => (!props.complementary ? 'none' : '2px solid #e91B89')};
    }

    &:disabled {
        color: #a5adba;
        background-color: rgba(9, 30, 66, 0.04);
        border: none;
    }
`;
