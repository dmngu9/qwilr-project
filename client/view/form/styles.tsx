import styled, { ThemedStyledProps, Theme } from '../../utils/styled-component';

import { ThemedButton } from '../themed-button';

export const Container = styled.div`
    width: 100%;
`;

export const ButtonContainer = styled.div`
    width: 100%;
    display: flex;
    justify-content: flex-end;
    margin-top: 20px;
`;

interface ExtraProps {
    rightMargin: boolean;
}

type Props = ThemedStyledProps<ExtraProps, Theme>;

export const SubmitButton = styled(ThemedButton)`
    min-width: 75px;
    margin-right: ${(props: Props) => (props.rightMargin ? '16px' : '0')};
`;

export const AlternativeActionButton = styled(ThemedButton)`
    min-width: 75px;
`;

export const Error = styled.div`
    font-size: 14px;
    color: red;
`;
