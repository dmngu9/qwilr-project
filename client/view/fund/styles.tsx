import styled from '../../utils/styled-component';
import { ThemedButton } from '../themed-button';

export const Fund = styled.div`
    font-size: 18px;
    font-weight: 300;
    color: #3a3a3a;
    padding: 0 0 10px;
    border-bottom: 1px solid ${props => props.theme.primaryColor};
`;

export const Deposit = styled.h3`
    font-size: 24px;
`;

export const ButtonContainer = styled.div`
    margin-top: 16px;
`;

export const ShowDepositFormButton = styled(ThemedButton)`
    margin-right: 16px;
    width: 80px;
`;

export const ShowWithdrawFormButton = styled(ThemedButton)`
    width: 80px;
`;
