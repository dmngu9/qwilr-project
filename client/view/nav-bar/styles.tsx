import styled, { css } from '../../utils/styled-component';
import { Link } from 'react-router-dom';

import { borderBox } from '../../utils/mixins';
import { ThemedButton } from '../themed-button';
import { media } from '../../utils/breakpoint';

export const Container = styled.div`
    ${media.phone`
        height: 64px;
        padding: 0 24px;
        font-size: 16px;
    `};

    height: 80px;
    width: 100%;
    position: fixed;
    top: 0;
    background-color: ${props => props.theme.primaryColor};
    display: flex;
    justify-content: flex-end;
    align-items: center;
    flex-wrap: wrap;
    padding: 0 64px;
    font-size: 24px;
    ${borderBox()};
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
`;

const underline = () => css`
    position: relative;

    &::before {
        position: absolute;
        content: '';
        bottom: 0;
        background-color: white;
        height: 2px;
        width: 50%;
        opacity: 0;
        transform: translateX(-100%);
        transition: transform 0.3s ease-in-out, opacity 0.3s ease-in-out;
    }

    &:hover::before {
        transform: translateX(0);
        opacity: 1;
    }
`;

export const StyledLink = styled(Link)`
    ${media.phone`
        margin-right: 24px;
    `};

    color: white;
    margin-right: 48px;
    text-decoration: none;
    ${underline()};
`;

export const LogOutButton = styled(ThemedButton)`
    ${media.phone`
        font-size: 16px;
    `};

    font-size: 24px;
    padding: 0;

    &:hover,
    &:active {
        color: white;
        background-color: transparent;
    }
    ${underline()};
`;
