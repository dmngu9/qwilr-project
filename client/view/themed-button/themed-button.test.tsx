import * as React from 'react';
import { shallow } from 'enzyme';
import { noop } from 'lodash';

import ThemedButton from './themed-button';
import { Button } from './styles';

describe('<ThemedButton/>', () => {
    it('should render correctly as primary button', () => {
        const wrapper = shallow(
            <ThemedButton type="submit" onClick={noop}>
                Submit
            </ThemedButton>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly as primary button when disabled', () => {
        const wrapper = shallow(
            <ThemedButton type="submit" onClick={noop} disabled>
                Submit
            </ThemedButton>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly as complementary button', () => {
        const wrapper = shallow(
            <ThemedButton type="submit" onClick={noop} complementary>
                Submit
            </ThemedButton>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly as complementary button when disabled', () => {
        const wrapper = shallow(
            <ThemedButton type="submit" onClick={noop} complementary disabled>
                Submit
            </ThemedButton>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should call onClick', () => {
        const mockCallback = jest.fn();
        const wrapper = shallow(
            <ThemedButton type="submit" onClick={mockCallback}>
                Submit
            </ThemedButton>
        );

        wrapper.find(Button).simulate('click');

        expect(mockCallback).toBeCalled();
    });
});
