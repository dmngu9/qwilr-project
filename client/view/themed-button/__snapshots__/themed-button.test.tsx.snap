// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemedButton/> should render correctly as complementary button 1`] = `
<styled.button
  complementary={true}
  onClick={[Function]}
  type="submit"
>
  Submit
</styled.button>
`;

exports[`<ThemedButton/> should render correctly as complementary button when disabled 1`] = `
<styled.button
  complementary={true}
  disabled={true}
  onClick={[Function]}
  type="submit"
>
  Submit
</styled.button>
`;

exports[`<ThemedButton/> should render correctly as primary button 1`] = `
<styled.button
  onClick={[Function]}
  type="submit"
>
  Submit
</styled.button>
`;

exports[`<ThemedButton/> should render correctly as primary button when disabled 1`] = `
<styled.button
  disabled={true}
  onClick={[Function]}
  type="submit"
>
  Submit
</styled.button>
`;
